x-pack/plugins/security_solution/server/lib/detection_engine/routes/utils.ts

```diff
@@ -104,10 +105,10 @@ export const transformBulkError = (
   ruleId: string,
   err: Error & { statusCode?: number }
 ): BulkError => {
-  if (Boom.isBoom(err)) {
+  if (err instanceof CustomBadRequestError) {
```

Comment:
I see a couple of places where we still use Boom. Should we also remove it from there?

```
❯ rg '@hapi/boom' ./x-pack/plugins/security_solution
./x-pack/plugins/security_solution/server/lib/detection_engine/routes/utils.ts
8:import Boom from '@hapi/boom';
```

And
```
❯ rg '@hapi/boom' ./packages
./packages/kbn-securitysolution-es-utils/BUILD.bazel
33:  "@npm//@hapi/boom",
40:  "@npm//@hapi/boom",

./packages/kbn-securitysolution-es-utils/src/transform_error/index.ts
9:import Boom from '@hapi/boom';

./packages/kbn-securitysolution-es-utils/src/transform_error/index.test.ts
9:import Boom from '@hapi/boom';

./packages/kbn-server-route-repository/BUILD.bazel
32:  "@npm//@hapi/boom",
41:  "@npm//@hapi/boom",

./packages/kbn-server-route-repository/src/decode_request_params.ts
12:import Boom from '@hapi/boom';
```

Reply:
Makes sense to remove all Boom usage as well. I'll get to it.

Reply:
@xcrzx 
So I've removed boom from `./packages/kbn-securitysolution-es-utils/src/transform_error/` files. `transformError` is indeed used within `x-pack/plugins/security_solution` and since we don't create any boom errors within the plugin with changes in this PR, it makes sense to remove boom from `transformError()`.

We can't remove `hapi/boom` from bazel files yet since it's used in `./packages/kbn-server-route-repository/src/decode_request_params.ts` which is used by APM team. I don't feel comfortable touching that bit of code so I'll leave it as it is.

90e4b504456f4733c2f01fb9fefd561805f68322

Reply:
So I reverted the changes to remove boom from `transformError`. It's being used in other plugins outside `security_solution` and causes test failures when removed. The changes in this PR are now within the scope of the tech. debt ticket.

Reply:
Ahh, okay. Well, so long as we do not create `Boom` errors in shared packages, I think it would be okay to remove `isBoom` checks from Security.